scripts/setup_for_wrf.py

```diff
@@ -31,98 +29,75 @@
 # load config file and create WRFConfig object
 wrf_config = load_wrf_config(configFile)
 
-# make a dict from WRFConfig object
-config = attrs.asdict(wrf_config)
-
-# parse start and end date
-try:
-    start_date = process_date_string(config["start_date"])
-    end_date = process_date_string(config["end_date"])
-
-    ## check that the dates are in the right order
-    assert end_date > start_date, "End date should be after start date"
-except Exception as e:
-    print("Problem parsing start/end times")
-    raise e
-
-
-# Perform some checks
-# Iteration count for filling variables
-# Next line moved to load_wrf_config()
-# assert iterationCount < 10, "Config key substitution exceeded iteration limit..."
-
 scripts = {}
 dailyScriptNames = ["run", "cleanup"]
-scriptNames = ["main", "run", "cleanup"]
-for scriptName in scriptNames:
-    templateScript = "{}_script_template".format(scriptName)
+script_names = ["main", "run", "cleanup"]
+script_paths = [wrf_config.main_script_template, wrf_config.run_script_template, wrf_config.cleanup_script_template]
+for script_name, script_path in zip(script_names, script_paths) :
     ## read the template run script
     assert os.path.exists(
-        config[templateScript]
-    ), "No template {} script was found at {}".format(
-        scriptName, config[templateScript]
-    )
-    try:
-        f = open(config[templateScript], "rt")
-        scripts[scriptName] = f.readlines()
+        script_path
+    ), f"No template script was found at {script_path}"
+    try :
+        f = open(script_path, "rt")
+        scripts[script_name] = f.readlines()
         f.close()
-    except Exception as e:
-        print("Problem reading in template {} script".format(scriptName))
+    except Exception as e :
+        print("Problem reading in template {} script".format(script_name))
         print(str(e))
 
-
-def decode_bytes(x):
-    if isinstance(x, bytes):
+def decode_bytes(x) :
+    if isinstance(x, bytes) :
         x = x.decode("utf-8")
     return x
 
 
-def purge(dir, pattern):
-    for f in os.listdir(dir):
-        if re.search(pattern, f) is not None:
+def purge(dir, pattern) :
```

Comment:
Can you ruff this file. It looks like there has been some weird non-ruff formatting applied

Reply:
That's the formatting from the "Reformat code" button in PyCharm that should be based on ruff. But it's obviously different, so maybe something is wrong with the plugin. It's probably better to just use the make command

Reply:
Have you configured the `ruff` pycharm plugin as well?